### DISCUSSION TITLE
Directories filters should try to show the distinct portions of paths too

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#16250] [Other Feature Feedback, Questions, & Ideas]
https://cs.github.com/github/docs?q=%22port%20122%22
<img width="813" alt="image" src="https://user-images.githubusercontent.com/2119212/167422362-a2e685b3-2364-46e4-b4c3-ac6e70dbd87a.png">

Here are the contents of the box on the right:
> Directories
‎translations/‎
‎content/admin/configuration/configuring-your-enterprise/‎
‎translations/zh-CN/content/admin/configuration/configuring-your-enterprise/‎
‎translations/ja-JP/content/admin/configuration/configuring-your-enterprise/‎
‎translations/pt-BR/content/admin/configuration/configuring-your-enterprise/‎
‎data/reusables/enterprise_installation/‎

Ideally, I'd see:

> Directories
‎translations/‎
‎content/...-your-enterprise/‎
‎translations/zh-CN/...-enterprise/‎
‎translations/ja-JP/...-enterprise/‎
‎translations/pt-BR/...-enterprise/‎
.../enterprise_installation/‎

**Comment by jsoref** [comment]
For the curious, the box is not tree oriented, you can actually see that as the first item after translations is `content/` whereas the next 3 aren't and the last is `data/`

Here's an example w/ the box expanded:
https://cs.github.com/github/docs?q=prodname_ghe_server%20language%3AMarkdown#
<img width="190" alt="image" src="https://user-images.githubusercontent.com/2119212/167423441-290d54f9-bae6-469d-8ce1-57ed8ebe46f4.png">

> Directories
‎translations/‎
‎data/reusables/‎
‎translations/zh-CN/‎
‎translations/ja-JP/‎
‎translations/pt-BR/‎
‎translations/zh-CN/data/reusables/‎
‎translations/ja-JP/data/reusables/‎
‎translations/pt-BR/data/reusables/‎
‎translations/es-ES/‎
‎translations/es-ES/data/reusables/‎
‎data/reusables/gated-features/‎
‎content/‎
‎translations/pt-BR/data/reusables/gated-features/‎
‎translations/zh-CN/data/reusables/gated-features/‎
‎translations/zh-CN/content/‎

A tree sort would have some value over this random sort...

> Directories
‎‎content/‎
‎data/reusables/‎
‎data/reusables/gated-features/‎
‎translations/‎
‎translations/es-ES/‎
‎translations/es-ES/data/reusables/‎
‎translations/ja-JP/‎
‎translations/ja-JP/data/reusables/‎
‎translations/pt-BR/‎
‎translations/pt-BR/data/reusables/‎
‎translations/pt-BR/data/reusables/gated-features/‎
‎translations/zh-CN/‎
‎translations/zh-CN/content/‎
‎translations/zh-CN/data/reusables/‎
‎translations/zh-CN/data/reusables/gated-features/‎

Note that `sort` on macOS does not result in a sort like this (it favors longer paths over shorter ones..., for reference, from painful experience, Google Sheets favors shorter strings over longer strings).

**Comment by colinwm** [comment]
Interesting observation! I intentionally tried to show the ends of the directory, since those are usually the most distinct. But in your case that's not true! Hmm...

**Reply by jsoref** [reply]
In a past life, I worked on a code search engine. (I've had quite a few past lives)

**Reply by jsoref** [reply]
As a general rule, localizations will ruin a basic heuristic like this.
As will some project schemas (hi java) with parallel `test` and `src` trees.

I'm actually still trying to develop a heuristic for something similar (automatically suggesting content to exclude from spell checking, which is generally for when there are a bunch of localization files that shouldn't be checked), so this is very much on my mind.

Here, it's really just the most common problem I run into when I search github/docs. Thankfully in this case, the results did offer translations, so I could exclude easily (and also the resulting regex was easy to hack to do what I wanted).
